rules/TypeDeclaration/Rector/ClassMethod/FalseReturnClassMethodToNullableRector.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Name;
+use PhpParser\Node\NullableType;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Return_;
+use PHPStan\Type\MixedType;
+use Rector\Core\Rector\AbstractRector;
+use Rector\PHPStanStaticTypeMapper\Enum\TypeKind;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\ClassMethod\FalseReturnClassMethodToNullableRector\FalseReturnClassMethodToNullableRectorTest
+ */
+final class FalseReturnClassMethodToNullableRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Change class method that returns false as invalid state, to nullable', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    /**
+     * @return false|int
+     */
+    public function run(int $number)
+    {
+        if ($number === 10) {
+            return false;
+        }
+
+        return $number;
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run(int $number): ?int
+    {
+        if ($number === 10) {
+            return null;
+        }
+
+        return $number;
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        // it already has a type
+        if ($node->returnType instanceof \PhpParser\Node) {
+            return null;
+        }
+
+        // there are 2 returns, one with false, other with something else except true
+        /** @var Return_[] $returns */
+        $returns = $this->betterNodeFinder->findInstanceOf($node, Return_::class);
```

Comment:
use `findInstancesOfInFunctionLikeScoped` instead to avoid inner function:

```suggestion
        $returns = $this->betterNodeFinder->findInstancesOfInFunctionLikeScoped($node, Return_::class);
```

Reply:
Done :+1: